File: spring-web/src/main/java/com/example/springwebex/service/PostEchoService.java
package com.example.springwebex.service;

import org.springframework.stereotype.Service;

import com.example.springwebex.model.restreq.BasicReqDto;
import com.example.springwebex.model.restresp.ResponseJsonDto;

import lombok.extern.slf4j.Slf4j;

@Service
@Slf4j
public class PostEchoService {

    public ResponseJsonDto<?> echo(BasicReqDto request) {
        ResponseJsonDto<BasicReqDto> response = new ResponseJsonDto<>();
        response.setResponse(request);

        return response;
    }

}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/service/PostEchoService.kt
package com.example.springwebex.service

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.stereotype.Service
import com.example.springwebex.model.restreq.BasicReqDto
import com.example.springwebex.model.restresp.ResponseJsonDto

@Service
class PostEchoService {
    private val log = KotlinLogging.logger {}
    fun <T : Any> echo(request: T): ResponseJsonDto<T> {
        val response = ResponseJsonDto<T>()
        response.response = request
        return response
    }
}
